src/Services/Azure/Authentication/CustomChainedCredential.cs

```diff
@@ -18,22 +19,72 @@ namespace AzureMcp.Services.Azure.Authentication;
 /// </remarks>
 public class CustomChainedCredential(string? tenantId = null) : TokenCredential
 {
-    private readonly ChainedTokenCredential _chainedCredential = CreateChainedCredential(tenantId);
+    public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        ChainedTokenCredential chainedCredential = CreateChainedCredential(tenantId);
+        return chainedCredential.GetToken(requestContext, cancellationToken);
+    }
+
+    public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        ChainedTokenCredential chainedCredential = CreateChainedCredential(tenantId);
+        return chainedCredential.GetTokenAsync(requestContext, cancellationToken);
+    }
+
+    private const string AuthenticationRecordEnvVarName = "AZURE_MCP_AUTHENTICATION_RECORD";
+    private const string OnlyUseBrokerCredentialEnvVarName = "AZURE_MCP_ONLY_USE_BROKER_CREDENTIAL";
+    private const string ClientIdEnvVarName = "AZURE_MCP_CLIENT_ID";
+
+    private static bool ShouldUseOnlyBrokerCredential()
+    {
+        string? onlyUseBrokerCredential = Environment.GetEnvironmentVariable(OnlyUseBrokerCredentialEnvVarName);
+        return onlyUseBrokerCredential == "true"
+            || onlyUseBrokerCredential == "True"
+            || onlyUseBrokerCredential == "T"
+            || onlyUseBrokerCredential == "1";
+    }
```

Comment:
nit: slightly more compact
```suggestion
    private static bool ShouldUseOnlyBrokerCredential() =>
        Environment.GetEnvironmentVariable(OnlyUseBrokerCredentialEnvVarName) switch
        {
            "true" => true,
            "True" => true,
            "T" => true,
            "1" => true,
            _ => false
        };
```

Reply:
This should be in a helper to be reused (if one doesn't already exist in .NET)

Reply:
I refactored it to a helper class method `GetEnvironmentVariableAsBool` that can be reused in other classes.